src/main/java/uk/gov/hmcts/reform/cwrdapi/service/impl/CaseWorkerServiceImpl.java

```diff
@@ -0,0 +1,279 @@
+        List<CaseWorkerProfile> caseWorkerProfiles = new ArrayList<>();
+        try {
+            getRolesAndUserTypes();
+            cwrsProfilesCreationRequest.forEach(cwrProfileCreationRequest -> {
+                CaseWorkerProfile caseWorkerProfileRespose = null;
+
+                caseWorkerProfile = caseWorkerProfileRepo
+                        .findByEmailId(cwrProfileCreationRequest.getEmailId());
+                if (Objects.isNull(caseWorkerProfile)) {
+
+                    caseWorkerProfileRespose = createCaseWorkerProfile(cwrProfileCreationRequest);
+                    if (Objects.nonNull(caseWorkerProfileRespose)) {
+                        // collecting all the successfully case worker profiles to save in caseworker db.
+                        caseWorkerProfiles.add(caseWorkerProfileRespose);
+
+                    }
+                } else if (Objects.nonNull(caseWorkerProfile) && !caseWorkerProfile.getDeleteFlag()) {
+
+                    //update the existing case worker profile logic
+                } else if (caseWorkerProfile.getDeleteFlag()) {
+
+                    // updating the status in idam to suspend.
+                }
+            });
+
+            /**  caseworker profile bacth save
+             in caseWorkerProfile request contains all the associated entities information and while
+             saving it automatically saves data in the sub entities like cwlocation, cwWorkArea,cwRole and
+             caseworker profile and no need to explicitly invoke the save method for each entities.
+             */
+            caseWorkerProfileRepo.saveAll(caseWorkerProfiles);
+            log.info("{}::case worker profiles inserted::" + caseWorkerProfiles.size(), loggingComponentName);
+
+        } catch (Exception exp) {
+
+            log.error("{}:: createCaseWorkerUserProfiles failed" + exp, loggingComponentName);
+        }
+        return ResponseEntity
+                .status(201)
+                .body(new CaseWorkerProfileCreationResponse("Case Worker Profiles Created."));
+    }
+
+    public CaseWorkerProfile createCaseWorkerProfile(CaseWorkersProfileCreationRequest cwrdProfileRequest) {
+
+        List<CaseWorkerLocation> cwLocations = new ArrayList<>();
+        List<CaseWorkerRole> caseWorkerRoles = new ArrayList<>();
+        List<CaseWorkerWorkArea> caseWorkerWorkAreas = new ArrayList<>();
+
+        //User Profile Call
+        ResponseEntity<Object> responseEntity = createUserProfileInIdamUP(cwrdProfileRequest);
+        if (responseEntity != null && responseEntity.getStatusCode().is2xxSuccessful()
+                && null != responseEntity.getBody()) {
+
+            UserProfileCreationResponse userProfileCreationResponse
+                    = (UserProfileCreationResponse) requireNonNull(responseEntity.getBody());
+            // case worker profile request mapping
+            caseWorkerProfile = new CaseWorkerProfile(userProfileCreationResponse.getIdamId(),
```

Comment:
Please use Setter Getters should be used instead of constructor for CaseWorkerProfile , Sonar warns if parameters count goes above 7 i think.

Reply:
done